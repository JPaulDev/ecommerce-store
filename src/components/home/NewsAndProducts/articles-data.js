import intel from '../../../../public/images/news-and-products/intel-article.webp';
import lovelace from '../../../../public/images/news-and-products/lovelace-article.webp';
import ryzen from '../../../../public/images/news-and-products/ryzen-article.webp';

const ARTICLES_DATA = [
  {
    title: 'NVIDIA 4000 Series Details Leaked',
    date: '29 Jun 2022',
    dateTime: '2022-06-29 13:00',
    image: lovelace,
    text: `NVIDIA's next-generation GeForce RTX 4090 "Ada" flagship graphics
    card allegedly scores over 19000 points in the 3DMark Time Spy Extreme
    synthetic benchmark, according to kopite7kimi, a reliable source with NVIDIA
    leaks. This would put its score around 66 percent above that of the current
    RTX 3090 Ti flagship.`,
    href: '/',
  },
  {
    title: 'Intel Misses Quarterly Financial Targets',
    date: '14 Jun 2022',
    dateTime: '2022-06-14 13:00',
    image: intel,
    text: `Intel Corporation today reported second-quarter 2022 financial
      results. "This quarter's results were below the standards we have set for
      the company and our shareholders. We must and will do better. The sudden
      and rapid decline in economic activity was the largest driver, but the
      shortfall also reflects our own execution issues," said Pat Gelsinger,
      Intel CEO.`,
    href: '/',
  },
  {
    title: 'Alleged Ryzen 7000 Series Benchmarks',
    date: '23 Jun 2022',
    dateTime: '2022-06-23 13:00',
    image: ryzen,
    text: `An AMD engineering sample CPU with the model number
      "100-000000593-20_Y" has appeared in a couple of graphics rendering tests,
      paired with an NVIDIA RTX A4000 GPU. The CPU appears to be a 6-core Ryzen
      7000-series chip that was fitted to a Gigabyte X670E Aorus Master
      motherboard.`,
    href: '/',
  },
];

export default ARTICLES_DATA;
