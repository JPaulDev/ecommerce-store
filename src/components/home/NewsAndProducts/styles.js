import styled from 'styled-components';

export const Section = styled.section`
  background-color: white;
  border: 1px solid ${({ theme }) => theme.colors.stone[200]};
  display: flex;
  margin-top: 15px;
  padding: 20px 25px;

  @media ${({ theme }) => theme.breakpoints.tabletSm} {
    padding: 30px 0;
  }

  @media ${({ theme }) => theme.breakpoints.desktopSm} {
    > article {
      flex: 1;
    }
  }
`;

export const Heading = styled.h2`
  font-family: 'Uni Sans Book', sans-serif;
  font-size: ${({ theme }) => theme.fontSizes[30]};
  margin-bottom: ${({ marginBottom }) => marginBottom};
  text-align: center;

  @media ${({ theme }) => theme.breakpoints.tabletSm} {
    text-align: left;
    margin-left: ${({ marginLeft }) => marginLeft};
  }
`;

export const ArticlesContainer = styled.div`
  display: flex;
  row-gap: 15px;
  flex-direction: column;

  > :last-child {
    border: 0;
    padding-bottom: 0;
  }

  @media ${({ theme }) => theme.breakpoints.tabletSm} {
    flex-direction: row;
  }
`;

export const Article = styled.article`
  flex: 1 1 250px;
  padding-bottom: 20px;
  display: flex;
  flex-direction: column;
  row-gap: 10px;
  border-bottom: 1px solid ${({ theme }) => theme.colors.stone[300]};
  align-items: center;

  > h3 {
    font-family: 'Uni Sans Book', sans-serif;
    font-size: ${({ theme }) => theme.fontSizes[18]};
    letter-spacing: 0.8px;
    text-align: center;
  }

  > time {
    color: ${({ theme }) => theme.colors.stone[500]};
    font-size: ${({ theme }) => theme.fontSizes[14]};
  }

  > p {
    font-size: ${({ theme }) => theme.fontSizes[15]};
    line-height: 1.7;
  }

  @media ${({ theme }) => theme.breakpoints.tabletSm} {
    align-items: flex-start;

    > h3 {
      text-align: left;
    }
  }

  @media ${({ theme }) => theme.breakpoints.tabletSm} {
    border-right: 1px solid ${({ theme }) => theme.colors.stone[300]};
    border-bottom: 0;
    padding: 0 25px;
  }
`;

export const Wrapper = styled.div`
  display: flex;
  height: 150px;
  width: 100%;
  justify-content: center;
  overflow: hidden;
  position: relative;

  > img {
    position: absolute;
  }
`;

export const Container = styled.div`
  border-left: 1px solid ${({ theme }) => theme.colors.stone[300]};
  padding: 0 25px;
  width: 30%;
  display: none;

  strong {
    font-family: 'Uni Sans Book', sans-serif;
    font-size: ${({ theme }) => theme.fontSizes[18]};
  }

  p {
    font-size: ${({ theme }) => theme.fontSizes[15]};
    line-height: 1.5;
    margin: 5px 0 15px;
  }

  > div {
    align-items: center;
    column-gap: 20px;
    display: flex;
    padding: 25px 0;
  }

  @media ${({ theme }) => theme.breakpoints.desktopSm} {
    display: block;
  }
`;
