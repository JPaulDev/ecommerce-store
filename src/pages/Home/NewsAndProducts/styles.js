import styled from 'styled-components';

export const Section = styled.section`
  background-color: white;
  border: 1px solid var(--color-grey-9);
  display: flex;
  margin-top: 15px;
  padding: 15px 25px 20px;

  @media screen and (min-width: 650px) {
    padding: 30px 0;
  }
`;

export const LeftContainer = styled.div`
  @media screen and (min-width: 1050px) {
    width: 68%;
  }
`;

export const Heading = styled.h2`
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.88rem;
  text-align: center;

  @media screen and (min-width: 650px) {
    text-align: left;
    margin-bottom: ${({ marginBottom }) => marginBottom};
    margin-left: ${({ marginLeft }) => marginLeft};
  }
`;

export const NewsContainer = styled.div`
  display: flex;

  & :last-child {
    border: 0;
    padding-bottom: 0;
  }

  @media screen and (max-width: 649px) {
    flex-direction: column;
  }
`;

export const ArticleBlock = styled.div`
  border-bottom: 1px solid #ddd;
  flex: 1 1 250px;
  padding-bottom: 20px;

  @media screen and (max-width: 649px) {
    a {
      justify-content: center;
    }
  }

  @media screen and (min-width: 650px) {
    border-right: 1px solid #ddd;
    border-bottom: 0;
    padding: 0 25px;
  }
`;

export const ArticleHeading = styled.h3`
  font-family: 'Uni Sans Book', sans-serif;
  font-size: 1.13rem;
  margin-top: 15px;
  text-align: center;

  @media screen and (min-width: 650px) {
    min-height: 50px;
    text-align: left;
    margin: 0;
  }
`;

export const Article = styled.article`
  font-size: 0.88rem;
  line-height: 1.7;
  margin: 10px 0;

  @media screen and (min-width: 650px) {
    text-align: left;
    margin-top: 15px;
  }
`;

export const Date = styled.div`
  color: var(--color-grey-8);
  font-size: 0.81rem;
  margin-bottom: 15px;

  @media screen and (max-width: 649px) {
    margin-bottom: 8px;
    text-align: center;
  }
`;

export const Image = styled.img`
  object-fit: cover;
  width: 100%;
`;

export const RightContainer = styled.div`
  border-left: 1px solid var(--color-grey-9);
  flex: 1;
  padding: 0 25px;
`;

export const Title = styled.strong`
  font-family: 'Uni Sans Book', sans-serif;
  font-size: 1.06rem;
`;

export const Text = styled.p`
  font-size: 0.88rem;
  margin: 6px 0 15px;
`;

export const Container = styled.div`
  align-items: center;
  column-gap: 18px;
  display: flex;
  padding: 25px 0;
`;
