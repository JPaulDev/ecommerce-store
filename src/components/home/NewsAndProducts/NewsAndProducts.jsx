import Image from 'next/future/image';
import hotSellers from '../../../../public/images/news-and-products/hot-sellers.webp';
import newArrivals from '../../../../public/images/news-and-products/new-arrivals.webp';
import { LinkChevron, LinkButton } from '../../ui';
import articles from './articles';
import * as Styled from './styles';

export default function NewsAndProducts() {
  const linkText = 'View All';

  return (
    <Styled.Section>
      <Styled.LeftContainer>
        <Styled.Heading marginBottom="20px" marginLeft="23px">
          Tech news & articles
        </Styled.Heading>
        <Styled.NewsContainer>
          {articles.map((article) => (
            <Styled.ArticleBlock key={article.title}>
              <Styled.ArticleHeading>{article.title}</Styled.ArticleHeading>
              <Styled.Date>{article.date}</Styled.Date>
              <Styled.Wrapper>
                <Image src={article.image} alt="" />
              </Styled.Wrapper>
              <Styled.Article>{article.text}</Styled.Article>
              <LinkChevron href={article.href} fontSize={13} iconSize={21}>
                Read Full Article
              </LinkChevron>
            </Styled.ArticleBlock>
          ))}
        </Styled.NewsContainer>
      </Styled.LeftContainer>
      <Styled.RightContainer>
        <Styled.Heading>Hot & New products</Styled.Heading>
        <Styled.Text>Check out our latest and greatest.</Styled.Text>
        <Styled.Container>
          <Image
            src={hotSellers}
            alt="An xbox gaming controller and an iPad."
          />
          <div>
            <Styled.Title>Our Hot Sellers</Styled.Title>
            <Styled.Text>See our most popular selling products.</Styled.Text>
            <LinkButton href="/" padding="12px 14px">
              {linkText}
            </LinkButton>
          </div>
        </Styled.Container>
        <Styled.Container>
          <Image
            src={newArrivals}
            alt="A widescreen computer monitor and virtual reality headset."
          />
          <div>
            <Styled.Title>Our Latest Arrivals</Styled.Title>
            <Styled.Text>
              When the competition wants to see what&lsquo;s new they check our
              website!
            </Styled.Text>
            <LinkButton href="/" padding="12px 14px">
              {linkText}
            </LinkButton>
          </div>
        </Styled.Container>
      </Styled.RightContainer>
    </Styled.Section>
  );
}
