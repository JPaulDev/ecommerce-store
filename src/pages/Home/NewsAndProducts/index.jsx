import useMediaQuery from '../../../hooks/useMediaQuery';
import hotSellers from '../../../assets/images/news-and-products/hot-sellers.webp';
import newArrivals from '../../../assets/images/news-and-products/new-arrivals.webp';
import articles from './articles';
import LinkWithChevron from '../../../components/LinkWithChevron';
import LinkButtonGrey from '../../../components/LinkButtonGrey';
import * as Styled from './styles';

export default function NewsAndProducts() {
  const isMatch = useMediaQuery('(min-width: 1050px)');

  return (
    <Styled.Section>
      <Styled.LeftContainer>
        <Styled.Heading marginBottom="20px" marginLeft="23px">
          Tech news & articles
        </Styled.Heading>
        <Styled.NewsContainer>
          {articles.map((article) => (
            <Styled.ArticleBlock key={article.title}>
              <Styled.ArticleHeading>{article.title}</Styled.ArticleHeading>
              <Styled.Date>{article.date}</Styled.Date>
              <Styled.Image
                src={article.image}
                width={630}
                height={150}
                alt=""
              />
              <Styled.Article>{article.text}</Styled.Article>
              <LinkWithChevron
                url={article.url}
                text="Read Full Article"
                fontSize="0.8rem"
                iconSize="21px"
              />
            </Styled.ArticleBlock>
          ))}
        </Styled.NewsContainer>
      </Styled.LeftContainer>
      {isMatch && (
        <Styled.RightContainer>
          <Styled.Heading>Hot & New products</Styled.Heading>
          <Styled.Text>Check out our latest and greatest.</Styled.Text>
          <Styled.Container>
            <img
              src={hotSellers}
              width={96}
              height={76}
              alt="An xbox gaming controller and an ipad"
            />
            <div>
              <Styled.Title>Our Hot Sellers</Styled.Title>
              <Styled.Text>See our most popular selling products.</Styled.Text>
              <LinkButtonGrey href="" padding="12px 14px">
                View All
              </LinkButtonGrey>
            </div>
          </Styled.Container>
          <Styled.Container>
            <img
              src={newArrivals}
              width={96}
              height={80}
              alt="A widescreen computer monitor and virtual reality headset"
            />
            <div>
              <Styled.Title>Our Latest Arrivals</Styled.Title>
              <Styled.Text>
                When the competition wants to see what&lsquo;s new they check
                our website!
              </Styled.Text>
              <LinkButtonGrey href="" padding="12px 14px">
                View All
              </LinkButtonGrey>
            </div>
          </Styled.Container>
        </Styled.RightContainer>
      )}
    </Styled.Section>
  );
}
