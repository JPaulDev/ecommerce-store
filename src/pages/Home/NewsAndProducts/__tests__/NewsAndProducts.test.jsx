import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NewsAndProducts from '../index';

describe('NewsAndProducts component', () => {
  it('renders a news section with 3 articles and latest products links', () => {
    const tree = renderer.create(<NewsAndProducts />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('NewsAndProducts component at different screen widths', () => {
  it('renders the latest products section 1150px and above', () => {
    window.resizeTo(1050, 768);
    render(<NewsAndProducts />);
    const heading = screen.getByRole('heading', {
      name: /hot & new products/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it('does not render the latest products section 1149px and below', () => {
    window.resizeTo(1049, 768);
    render(<NewsAndProducts />);
    const heading = screen.queryByRole('heading', {
      name: /hot & new products/i,
    });

    expect(heading).not.toBeInTheDocument();
  });
});
