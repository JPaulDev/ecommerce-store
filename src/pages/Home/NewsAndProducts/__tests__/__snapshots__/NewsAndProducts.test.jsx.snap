// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsAndProducts component renders a news section with 3 articles and latest products links 1`] = `
<section
  className="sc-hKMtZM eYsiwP"
>
  <div
    className="sc-eCYdqJ ckHFvq"
  >
    <h2
      className="sc-jSMfEi evCgse"
    >
      Tech news & articles
    </h2>
    <div
      className="sc-gKXOVf bKsfLq"
    >
      <div
        className="sc-iBkjds gypTwr"
      >
        <h3
          className="sc-ftvSup bljiMK"
        >
          NVIDIA 4000 Series Details Leaked
        </h3>
        <div
          className="sc-jqUVSM cpsnxr"
        >
          29 Jun 2022
        </div>
        <img
          alt=""
          className="sc-kDDrLX hQnLxe"
          height={150}
          src="lovelace-article.webp"
          width={630}
        />
        <article
          className="sc-papXJ gOwmkX"
        >
          NVIDIA's next-generation GeForce RTX 4090 "Ada" flagship graphics
    card allegedly scores over 19000 points in the 3DMark Time Spy Extreme
    synthetic benchmark, according to kopite7kimi, a reliable source with NVIDIA
    leaks. This would put its score around 66 percent above that of the current
    RTX 3090 Ti flagship.
        </article>
        <a
          className="sc-gsnTZi iWfFcv"
          data-testid="link-chevron"
          href=""
        >
          Read Full Article
          <i
            className="sc-bczRLJ dxnJQy material-icons"
          >
            chevron_right
          </i>
        </a>
      </div>
      <div
        className="sc-iBkjds gypTwr"
      >
        <h3
          className="sc-ftvSup bljiMK"
        >
          Intel Misses Quarterly Targets
        </h3>
        <div
          className="sc-jqUVSM cpsnxr"
        >
          14 Jun 2022
        </div>
        <img
          alt=""
          className="sc-kDDrLX hQnLxe"
          height={150}
          src="intel-article.webp"
          width={630}
        />
        <article
          className="sc-papXJ gOwmkX"
        >
          Intel Corporation today reported second-quarter 2022 financial
      results. "This quarter's results were below the standards we have set for
      the company and our shareholders. We must and will do better. The sudden
      and rapid decline in economic activity was the largest driver, but the
      shortfall also reflects our own execution issues," said Pat Gelsinger,
      Intel CEO.
        </article>
        <a
          className="sc-gsnTZi iWfFcv"
          data-testid="link-chevron"
          href=""
        >
          Read Full Article
          <i
            className="sc-bczRLJ dxnJQy material-icons"
          >
            chevron_right
          </i>
        </a>
      </div>
      <div
        className="sc-iBkjds gypTwr"
      >
        <h3
          className="sc-ftvSup bljiMK"
        >
          Alleged Ryzen 7000 Series Benchmarks
        </h3>
        <div
          className="sc-jqUVSM cpsnxr"
        >
          23 Jun 2022
        </div>
        <img
          alt=""
          className="sc-kDDrLX hQnLxe"
          height={150}
          src="ryzen-article.webp"
          width={630}
        />
        <article
          className="sc-papXJ gOwmkX"
        >
          An AMD engineering sample CPU with the model number
      "100-000000593-20_Y" has appeared in a couple of graphics rendering tests,
      paired with an NVIDIA RTX A4000 GPU. The CPU appears to be a 6-core Ryzen
      7000-series chip that was fitted to a Gigabyte X670E Aorus Master
      motherboard.
        </article>
        <a
          className="sc-gsnTZi iWfFcv"
          data-testid="link-chevron"
          href=""
        >
          Read Full Article
          <i
            className="sc-bczRLJ dxnJQy material-icons"
          >
            chevron_right
          </i>
        </a>
      </div>
    </div>
  </div>
</section>
`;
